app/components/sbom/app-list/header/index.ts

```diff
@@ -0,0 +1,91 @@
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import IntlService from 'ember-intl/services/intl';
+import MeService from 'irene/services/me';
+import OrganizationService from 'irene/services/organization';
+import ENUMS from 'irene/enums';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import styles from './index.scss';
+
+interface PlatformObject {
+  key: string;
+  value: number;
+}
+
+interface SbomAppListHeaderArgs {
+  packageNameSearchQuery: string;
+  selectedPlatformValue: number;
+  onQueryChange(event: Event): void;
+  handleClear(): void;
+  filterPlatform(platform: PlatformObject): void;
+}
+
+export default class SbomAppListHeaderComponent extends Component<SbomAppListHeaderArgs> {
+  @service declare intl: IntlService;
+  @service declare me: MeService;
+  @service declare organization: OrganizationService;
+
+  @tracked selectedPlatform = {
+    key: 'All',
+    value: -1,
+  };
+
+  constructor(owner: unknown, args: SbomAppListHeaderArgs) {
+    super(owner, args);
+
+    if (this.args.selectedPlatformValue) {
+      for (const platformObject of this.platformObjects) {
+        if (platformObject.value === Number(this.args.selectedPlatformValue)) {
+          this.selectedPlatform = platformObject;
+          break;
+        }
+      }
+    }
```

Comment:
make same changes as this comment https://github.com/appknox/irene/pull/1331#discussion_r1479296727